### PR TITLE
Add Jest and Testing-Library in Typescript

### PR DESCRIPTION
**Pull request by wesleylhandy**
I recently downloaded your starter and found my self needing to add unit tests. I figured adding this setup would be rather simple and yet super helpful for anyone using this in the future. 

I added the latest version of `jest` and `@testing-library`, as well as the corresponding types and linting rules.

Please note the addition of `tsconfig.spec.json` in the root. This is just to override the `jsx` rule for the testing environment. Otherwise, tests will get very weird errors that are not easy to track down without experience.

Besides adding testing rules, I did make a few other changes in `.eslintrc`:

The following rules were conflicting locally, since both `es-lint` and `@typescript-eslint` have versions:

```json
    "import/no-unresolved": 0,
    "quotes": "off",
```

I turned these off in the `overrides`. 

I also added one rule to allow unused variables to be prefixed with an underscore (helpful in mocking, and unused props):

```json
    "@typescript-eslint/no-unused-vars": [2, { "argsIgnorePattern": "^_" }]
```

### CODE REVIEW

**Comment by jpedroschmitz** [comment]
Hey @wesleylhandy, thanks a lot for this PR! I had tests in my backlog here but never had a chance to include them.

I'll be taking a look into this over the weekend 🚀

File: package.json [hunk 1]
```diff
@@ -27,34 +28,47 @@
     ]
   },
   "dependencies": {
+    "@testing-library/react": "^11.2.7",
+    "@testing-library/react-hooks": "^7.0.0",
```
Comment (jpedroschmitz) [inline package.json:32]: We can move these two deps to `devDependencies`.

File: src/types/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export interface BaseComponentProps {
+  children?: unknown;
+}
```
Comment (jpedroschmitz) [inline src/types/index.ts:3]:
```suggestion
import { ReactNode } from 'react';

export interface BaseComponentProps {
  children?: ReactNode;
}
```

File: src/components/Title.tsx [hunk 3]
```diff
@@ -1,5 +1,20 @@
 import React from 'react';
+import { BaseComponentProps } from '@/types';
 
-const Title: React.FC = () => <h1>Hello TypeScript!</h1>;
+export interface TitleProps extends BaseComponentProps {
+  testId?: string;
```
Comment (jpedroschmitz) [inline src/components/Title.tsx:5]: We can remove this `testId` logic. See the other comment for more info.

File: src/components/Title.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  it(`renders a Title component`, () => {
+    renderTitle();
+
+    expect(screen.getByTestId(`title`)).toBeInTheDocument();
+  });
+
+  it(`renders text children`, () => {
+    renderTitle({ children: `Test Title` });
+
+    expect(screen.getByTestId(`title`)).toHaveTextContent(`Test Title`);
```
Comment (jpedroschmitz) [inline src/components/Title.spec.tsx:21]: I usually don't like to use `data-testid`, unless there's no option. 
You could do something like this here:
```
expect(getByText('Test Title')).toBeInTheDocument()
```

**Review by jpedroschmitz — CHANGES_REQUESTED** [review]
Great job @wesleylhandy! I left a few comments 👍🏻 

It would be great if you could also fix the conflicts!

File: package.json (hunk 1 above)
Comment (wesleylhandy) [reply package.json]: Just moved. I also updated `.eslintrc` to not trigger error when importing `devDependencies` into test and spec files.

File: src/types/index.ts (hunk 2 above)
Comment (wesleylhandy) [reply src/types/index.ts]: Done

File: src/components/Title.tsx (hunk 3 above)
Comment (wesleylhandy) [reply src/components/Title.tsx]: Done

File: src/components/Title.spec.tsx (hunk 4 above)
Comment (wesleylhandy) [reply src/components/Title.spec.tsx]: Done

**Comment by wesleylhandy** [comment]
When I rebased to fix the merge conflicts, I lost all my original commit information. Maybe I should have cherry-picked or squashed before rebasing, sorry.

**Comment by jpedroschmitz** [comment]
Don't worry. We can squash and merge and have a better commit message!

**Review by jpedroschmitz — APPROVED** [review]
LGTM! Thanks for all the adjustments ❤️

**Pull request merged**